### PR TITLE
Migrate to androidx

### LINKED ISSUE #5260: [androidx] Investigate running Jetifier over older SDK versions
If we could migrate older SDKs to AndroidX, this would let us keep them when we eventually include a new SDK version that uses AndroidX.

We don't plan to ship AndroidX support with SDK 35 but want to start investigating now.

### PR DESCRIPTION
**Pull request by Szymon20000**
# Why

resolves: https://github.com/expo/expo/issues/5260

# How

https://developer.android.com/jetpack/androidx/migrate
https://blog.danlew.net/2018/11/14/the-reality-of-migrating-to-androidx/
https://gist.github.com/loganj/7535a13e98be83460f362b63dbd13e07

# Test Plan

Run NCL, native tests (androidTest,  test)

[update]
I've made rebase and tested it once again.

### CODE REVIEW

**Review by mczernek — APPROVED** [review]

**Review by ide — CHANGES_REQUESTED** [review]
This is a great start and will make it so much smoother for Expo developers on Android when we add RN 0.61 to the Expo client.

Do we need to recompile any AARs, such as ReactAndroid for SDK 33 and 34?

**Release timing:** I think the best time to land this commit will be after the SDK 35 branch has been cut for a few reasons. SDK 35 uses RN 0.59.8, which does not have AndroidX support outside of our changes. So if someone uses ExpoKit v35 and looks for a module that is compatible with RN 0.59.8 (no AndroidX), we want their app to compile.

Additionally, this is a complex PR and the longer we have to test it, the better. By merging it after the SDK 35 branch has been created, this maximizes the amount of time we can internally test this PR on master before creating the SDK 36 branch.

**Testing:** We need a thorough test plan as part of this PR, especially because it has to be so large in several ways unfortunately. Some of the activity screens like the OAuth webview activity seem important to test since they reference class names from XML. We also need to test loading projects that use older SDK versions.

**Vendored libraries:** what should people who are updating vendored libraries like Maps be aware of? When @brentvatne or @sjchmiela updates GestureHandler, for example, will the new versions have AndroidX support or will they need to make manual changes, in which case what should they do? Do we need to reach out to library authors to have a sustainable path forward?

File: android/app/build.gradle [hunk 1]
```diff
@@ -108,14 +108,14 @@ apply from: "../../node_modules/react-native-unimodules/gradle.groovy"
 dependencies {
   implementation fileTree(dir: 'libs', include: ['*.jar'])
 
-  implementation 'com.android.support:multidex:1.0.1'
+  implementation 'androidx.multidex:multidex:2.0.0'
 
   // Our dependencies
-  implementation "com.android.support:appcompat-v7:$supportLibVersion"
+  implementation 'androidx.appcompat:appcompat:1.0.0'
```
[the remark this answers is not in the document]
Comment (bbarthec) [reply android/app/build.gradle]: Take a look here: https://developer.android.com/jetpack/androidx/versions
Version of each package seems to be diverged 🤔

File: android/app/src/main/res/layout/error_activity_new.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.v4.view.ViewPager
+<androidx.viewpager.widget.ViewPager
```
[the remark this answers is not in the document]
Comment (Szymon20000) [reply android/app/src/main/res/layout/error_activity_new.xml]: I checked this activity and it works.

File: android/cameraview/cameraview/src/main/api14/com/google/android/cameraview/Camera1.java [hunk 3]
```diff
@@ -22,7 +22,7 @@
 import android.media.CamcorderProfile;
 import android.media.MediaRecorder;
 import android.os.Build;
-import android.support.v4.util.SparseArrayCompat;
+import androidx.collection.SparseArrayCompat;
```
[the remark this answers is not in the document]
Comment (Szymon20000) [reply android/cameraview/cameraview/src/main/api14/com/google/android/cameraview/Camera1.java]: I think so. However, @bbarthec is more familiar with this topic.

**Review by ide — APPROVED** [review]

**Pull request merged**